lib/sql.js

```diff
@@ -603,11 +603,25 @@ SQLConnector.prototype.updateAttributes = function(model, id, data, options, cb)
  * @private
  */
 SQLConnector.prototype.replaceById = function(model, id, data, options, cb) {
+  var self = this;
   if (!isIdValuePresent(id, cb)) return;
   var where = this._buildWhereObjById(model, id, data);
-  this._replace(model, where, data, options, cb);
+  this.find(model, where.id, options, function(err, foundInstance) {
```

Comment:
This will send a new SQL query and create a race condition - what if some other client calls create or delete on a record with the same `id`?

I think we need to fix `self._replace` instead so that it parses the response from the SQL backend and detect when no record was updated. 
I think we need to call `self._replace` and 

@Amir-61 are you sure there is a single solution that works for all SQL databases? I would personally expect getting hold of the number of updated rows is database/driver specific.